src/services/telemetryService.ts

```diff
@@ -115,34 +29,167 @@ interface TelemetryOptions {
-	sendTelemetryException(gitpodHost: string, error: Error, properties?: TelemetryEventProperties): void;
+	sendTelemetryEvent(eventName: string, properties?: TelemetryEventProperties): void;
+	sendTelemetryException(error: Error, properties?: TelemetryEventProperties): void;
 
-	sendUserFlowStatus(status: string, flow: UserFlowTelemetry): void;
+	sendUserFlowStatus(status: string, flow: UserFlowTelemetryProperties): void;
 }
 
-export class TelemetryService extends BaseTelemetryReporter implements ITelemetryService {
-	constructor(extensionId: string, extensionVersion: string, segmentKey: string, logger: vscode.LogOutputChannel) {
-		super(extensionId, extensionVersion, {
-			release: os.release(),
-			platform: os.platform(),
-			architecture: os.arch(),
-		}, gitpodHost => analyticsClientFactory(gitpodHost, segmentKey, logger));
+const TRUSTED_VALUES = new Set([
+	'gitpodHost'
+]);
+
+export class TelemetryService extends Disposable implements ITelemetryService {
+	private analitycsClients: Map<string, Analytics> = new Map();
+	private telemetryLogger: vscode.TelemetryLogger;
+
+	constructor(segmentKey: string, private readonly logService: ILogService) {
+		super();
+
+		this.telemetryLogger = this._register(vscode.env.createTelemetryLogger(
+			{
+				sendEventData: (eventName, data) => {
+					const idx = eventName.indexOf('/');
+					eventName = eventName.substring(idx + 1);
+
+					const properties = data ?? {};
+
+					const gitpodHost: string | undefined = properties['gitpodHost'];
+					if (!gitpodHost) {
+						logService.error(`Missing 'gitpodHost' property in event ${eventName}`);
+						return;
+					}
+
+					delete properties['gitpodHost'];
+
+					this.getSegmentAnalyticsClient(gitpodHost, segmentKey)?.track({
+						anonymousId: vscode.env.machineId,
+						event: eventName,
+						properties
+					}, (err) => {
+						if (err) {
+							logService.error('Failed to log event to app analytics:', err);
+						}
+					});
+				},
+				sendErrorData: (error, data) => {
+					const properties = data ?? {};
+
+					// Unhandled errors have no data so use host from config
+					const gitpodHost = properties['gitpodHost'] ?? Configuration.getGitpodHost();
+					const errorMetricsEndpoint = this.getErrorMetricsEndpoint(gitpodHost);
+
+					properties['error_name'] = error.name;
+					properties['error_message'] = error.message;
+					properties['debug_workspace'] = String(properties['debug_workspace'] ?? false);
+
+					const workspaceId = properties['workspaceId'] ?? '';
+					const instanceId = properties['instanceId'] ?? '';
+					const userId = properties['userId'] ?? '';
+
+					delete properties['gitpodHost'];
+					delete properties['workspaceId'];
+					delete properties['instanceId'];
+					delete properties['userId'];
+
+					const jsonData = {
+						component: 'vscode-desktop-extension',
+						errorStack: error.stack || String(error),
```

Comment:
Error stack is full now, is it expected? I thought we will filter user data out first at before, i.e. user home path/email?

Reply:
i.e. deleted `telemetry.ts` file methods `anonymizeFilePaths` and `removePropertiesWithPossibleUserInfo`